src/transformers/models/siglip/convert_siglip_to_hf.py

```diff
@@ -266,32 +277,49 @@ def convert_siglip_checkpoint(model_name, pytorch_dump_folder_path, verify_logit
 
     # get vocab file
     if "i18n" in model_name:
-        vocab_file = "/Users/nielsrogge/Documents/SigLIP/multilingual_vocab/sentencepiece.model"
+        vocab_file = "/Users/mervenoyan/Desktop/siglips/mc4/sentencepiece.model"
     else:
-        vocab_file = "/Users/nielsrogge/Documents/SigLIP/english_vocab/sentencepiece.model"
+        vocab_file = "/Users/mervenoyan/Desktop/siglips/c4_en/sentencepiece.model"
 
     # load original state dict
     data = load(checkpoint)
     state_dict = flatten_nested_dict(data)
 
-    # remove and rename some keys
+    if (
+        model_name == "siglip-so400m-patch16-256-i18n"
+    ):  # make state dict compatible with rest of the SiglIPs, add param/ prefix and encoderblock index
+        new_state_dict = {}
+        for k, v in state_dict.items():
+            if "img/Transformer/encoderblock/" in k:
+                original_array = np.array(v)
+                split_arrays = np.array_split(original_array, config.vision_config.num_hidden_layers, axis=0)  # 27
+                for i in range(config.vision_config.num_hidden_layers):
+                    new_key = re.sub(r"(encoderblock)/", rf"\1_{i}/", k)
+                    new_state_dict[f"params/{new_key}"] = split_arrays[i].squeeze()
+            else:
+                new_state_dict[f"params/{k}"] = v
+
+    state_dict = new_state_dict
```

Comment:
let's add this to the new conversion script!

Reply:
@ArthurZucker architecture is not different here, it's only the way they put the encoder weights together which needed a modification before renaming the keys.